### PR TITLE
eof: Disable `gasTests` for EOF

### PR DESCRIPTION
**Pull request by rodiazet**
`gasTests` are supposed to be run not via IR. As long as they are run this way they cannot be compiled to EOF. 
- Make `GasTest` supports `bytecodeFormat` flag and mark them all to run only for `current` EVM version
- Enable excluded earlier `gasTests` in CI testing for EOF.
- Add `bytecodeFormat: legacy` flag to `gasTests`

### CODE REVIEW

File: .circleci/soltest.sh [hunk 1]
```diff
@@ -59,19 +59,6 @@ EOF_EXCLUDES=(
...
-    --run_test='!gasTests/dispatch_large'
-    --run_test='!gasTests/dispatch_large_optimised'
-    --run_test='!gasTests/dispatch_medium'
-    --run_test='!gasTests/dispatch_medium_optimised'
-    --run_test='!gasTests/dispatch_small'
-    --run_test='!gasTests/dispatch_small_optimised'
-    --run_test='!gasTests/exp'
-    --run_test='!gasTests/exp_optimized'
-    --run_test='!gasTests/storage_costs'
-    --run_test='!yulStackLayout/literal_loop'
```
Comment (cameel) [inline .circleci/soltest.sh:71]: This one's not a gas test. I guess it's included because it pases now? I'd still at least put it in a separate commit, because it does not match the current description.

File: test/libsolidity/GasTest.h [hunk 2]
```diff
@@ -32,7 +32,7 @@
 namespace solidity::frontend::test
 {
 
-class GasTest: AnalysisFramework, public TestCase
+class GasTest: AnalysisFramework, public EVMVersionRestrictedTestCase
```
Comment (cameel) [inline test/libsolidity/GasTest.h:35]: If we're doing this, then we should also convert this bit: https://github.com/ethereum/solidity/blob/82cefdf3374ed610614054290933f6300265c6f8/test/libsolidity/GasTest.cpp#L118-L120

I.e. just remove it - `AnalysisFramework` by default configures the `ComilerStack` to use EVM and EOF versions from command-line. Then in the tests also include `EVMVersion: =current` to explicitly restrict them to the latest EVM version (unless they do work with a wider range of versions but I doubt it, and it's probably not worth it to bother checking).

Without it the values read from the test are actually ignored. I actually wonder why they even pass here given that only EVM version is being overridden and EOF is left unchanged.

**Review by cameel — COMMENTED** [review]
For now disabling those on EOF is ok. But the PR needs a few adjustments because the way it is now, it seems to me that the EVM and EOF versions from the test settings will get ignored. I'm actually not sure how these tests even pass without that.

Eventually we should either make them work via-IR with and without EOF or just get rid of them if they're not worth converting. But that's something for later.

File: .circleci/soltest.sh (hunk 1 above)
Comment (rodiazet) [reply .circleci/soltest.sh:71]: I think I just removed to many lines. This test lagacy `delegatecall` so it won't pass.

File: test/libsolidity/GasTest.h (hunk 2 above)
Comment (rodiazet) [reply test/libsolidity/GasTest.h:35]: They did not pass before the change. Now they pass because this tests are omitted for EOF compiler flag. 
Applied the change you suggested.

**Review by cameel — APPROVED** [review]

File: test/libsolidity/GasTest.h (hunk 2 above)
Comment (cameel) [reply test/libsolidity/GasTest.h:35]: Ah, I somehow keep forgetting that EOF is not included by default :)

**Review by shahab136586 — COMMENTED** [review]
alphawallet
